File: src/2894-Divisible-and-Non-divisible-Sums-Difference/divisible_and_non_divisible_sums_difference.cpp
#include <iostream>

class Solution {
public:
    int differenceOfSums(int n, int m) {
        int num1 = 0;
        int num2 = 0;

        for (int i = 1; i <= n; i++) {
            if (i % m == 0) {
                num2 += i;
            }
            else {
                num1 += i;
            }
        }

        return num1 - num2;
    }
};


int main() {
    int m = 10;
    int n = 3;
    Solution solution;

    std::cout << solution.differenceOfSums(n, m);
    
    return 0;
}

File: src/2894-Divisible-and-Non-divisible-Sums-Difference/divisible_and_non_divisible_sums_difference.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        num1: int = 0
        num2: int = 0
        i: int = 1

        while i <= n:
            if i % m == 0:
                num2 += i
            else:
                num1 += i
            i += 1

        return num1 - num2

if __name__ == "__main__":
    n: int = 10
    m: int = 3
    solution = Solution()
    
    print(solution.differenceOfSums(n, m))